src/app/modules/search-input/services/advanced-search-autocomplete.service.ts

```diff
@@ -81,17 +82,22 @@ export class AdvancedSearchAutocompleteService<T> {
     if (
       /^["'].*["']?$/.test(currentQuery[from - 1]) && /^["'].*["']?$/.test(currentQuery[to])
     ) {
-      updatedValue = updatedValue.replace(/['"]/g, '');
+      updatedValue = updatedValue.replace(/^["'](.*)["']$/, '$1');
```

Comment:
1. I've noticed that the regular expression `^["'](.*)["']$` is used to detect a string surrounded by double quotes `"` and single quotes `'` at the same time, giving a false success for mixed qoutes `"test'`. I'd suggested to have 2 separate regular expressions for single quotes `'` and double quote `"`.
2. Also, looks like same regular expression `^["'](.*)["']$` is used to detect a string surrounded by quotes giving a false success for `"test"test"`. I've suggested to replace `.*` with `[^"]` for double quotes and with `[^']` for single quotes, so that the string could not contain a closing quote. 
3. It might be reasonable to extract the regular expression into a reusable constant
4. Not completely sure about this, but we could update tests to include a couple of invalid user's input similar to the examples in "1" and "2"

Reply:
I've updated code, in this case autocompletion will not work if you try to wrap it in unmatched quotes like:
'hello"
And query will be invalid without transformations:

double quotes -- works:
<img width="365" alt="Screenshot 2023-12-04 at 12 11 03" src="https://github.com/truenas/webui/assets/22980553/3ff6f197-739f-4e14-89be-2f02aded0824">

single quotes -- works:
<img width="370" alt="Screenshot 2023-12-04 at 12 11 09" src="https://github.com/truenas/webui/assets/22980553/011407d5-7eeb-47f9-a7f0-468db0502a70">

mixed quotes -- error and no suggestions:
<img width="345" alt="Screenshot 2023-12-04 at 12 11 38 1" src="https://github.com/truenas/webui/assets/22980553/6db3d2f1-84f9-4d70-a649-8be781205f82">

"test"test" -- will also fail and blocks transformations

So the transformation method is only called once the query counts as VALID.
Autocomplete suggests when quotes are matching, I don't think that user will do something like **"hello'** or **"he"llo"** -- we have auto-closing quotes which will enforce this should not happen.